### PR TITLE
Add GlobalMixin types for Application plugins

### LINKED ISSUE #7141: Missing Application plugins & Graphics' lineStyle types
I just updated to PixiJS v6 RC and wanted to check if the updated types in v6 broke our Angular project and I get this errors that certain attributes don't exist anymore on certain types. If I check the v6 docs, the attributes are still there and not deprecated. Then I saw, that the pixi.js.d.ts is missing from node_modules/pixijs and maybe that confuses the typescript compiler.

I get this error from VS Code:

`"Unable to open 'pixi.js.d.ts': Unable to read file '/Users/toms/Coding/WaveModeler/node_modules/pixi.js/pixi.js.d.ts' (Error: Unable to resolve non-existing file '/Users/toms/Coding/WaveModeler/node_modules/pixi.js/pixi.js.d.ts')."`

And a lot of compilation errors like this:
`ERROR in src/app/main/processTabContent/editor/editor.component.ts:102:18 - error TS2339: Property 'loader' does not exist on type 'Application'.
this.pixiApp.loader.load(() => this.pixiApp.renderer.render(this.pixiApp.stage));`

### Environment

- **`pixi.js` version**: 6.0.0-rc
- **OS & Version**: macOS 11.1 Big Sur
- **Angular Version**: Angular 10
- **Typescript Version**: 3.9.7

### PR DESCRIPTION
**Pull request by bigtimebuddy**
Partially fixes #7141 by adding types for Application plugins. Will support types for `loader`, `resizeTo` and `ticker`.

### CODE REVIEW

File: packages/app/src/Application.ts [hunk 1]
```diff
@@ -10,12 +10,11 @@ export interface IApplicationPlugin {
...
-    autoStart?: boolean;
-    sharedTicker?: boolean;
-    sharedLoader?: boolean;
-    resizeTo?: Window | HTMLElement;
-}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IApplicationOptions extends IRendererOptionsAuto, GlobalMixins.IApplicationOptions {}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface Application extends GlobalMixins.DisplayObject {}
```
Comment (Zyie) [inline packages/app/src/Application.ts:17]: I think this supposed to be `GlobalMixins.Application` that is declared in the loaders package?

**Review by Zyie — CHANGES_REQUESTED** [review]

File: packages/app/src/Application.ts (hunk 1 above)
Comment (bigtimebuddy) [reply packages/app/src/Application.ts:17]: Good catch :+1:

**Review by Zyie — APPROVED** [review]

**Review by themoonrat — APPROVED** [review]

**Comment by zakjan** [comment]
Hi @bigtimebuddy, after upgrading from 6.0.0-rc to 6.0.0-rc.2, I started getting TS errors with PIXI.Application constructor, that my options object is missing required options sharedLoader, autoStart, sharedTicker. The options should be optional, right?

**Comment by Zyie** [comment]
Hey @zakjan this has been addressed in this PR #7151

**Comment by zakjan** [comment]
Perfect, thanks!

**Pull request merged**